torch/distributed/tensor/parallel/style.py

```diff
@@ -388,3 +325,67 @@ def make_output_reshard_tensor(
     """
 
     return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined]
+
+
+class RowwiseParallel(ParallelStyle):
+    """
+    Partitioning the row of a module.
+    We assume the input to be a sharded :class:`DTensor` and output to be a replicated :class:`DTensor`.
+    """
+
+    def __init__(self, _prepare_input=make_input_shard_1d_last_dim, _prepare_output=make_output_replicate_1d) -> None:
+        super().__init__(_prepare_input, _prepare_output)
+
+
+class ColwiseParallel(ParallelStyle):
+    """
+    Partitioning the column of a tensor or module.
+    We assume the input to be a replicated :class:`DTensor` and output to be a sharded :class:`DTensor`.
+    """
+
+    def __init__(self, _prepare_input=make_input_replicate_1d, _prepare_output=make_output_replicate_1d) -> None:
+        super().__init__(_prepare_input, _prepare_output)
+
+
+class ColwiseParallelForPairwise(ColwiseParallel):
+    """
+    Partitioning the column of a tensor or module.
+    We assume the input to be a replicated :class:`DTensor` and output to be a sharded :class:`DTensor`.
+    We don't perform reshard for output.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(make_input_replicate_1d, None)
+
+
+class RowwiseParallelForPairwise(RowwiseParallel):
+    """
+    Partitioning the row of a module.
+    We assume the input to be a sharded :class:`DTensor` and output to be a replicated :class:`DTensor`.
+    We don't perform reshard for input.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(None, make_output_replicate_1d)
+
+
+class ColwiseParallelForAttn(ColwiseParallel):
+    """
+    Partitioning the column of a tensor or module.
+    We assume the input to be a replicated :class:`DTensor` and output to be a sharded :class:`DTensor`.
+    We don't perform reshard for output.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(make_input_replicate_1d, make_sharded_output_tensor)
+
+
+class RowwiseParallelForAttn(RowwiseParallel):
```

Comment:
Let's try to make sure we only have `RowwiseParallel` and `ColwiseParallel`, and `PairwiseParallel`, I feel adding a new style with subtle difference on input/output might not be a good UX.